x-pack/plugins/security_solution/server/lib/telemetry/sender.ts

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { cloneDeep } from 'lodash';
+
+import { LegacyAPICaller } from 'kibana/server';
+import { Logger, CoreStart } from '../../../../../../src/core/server';
+import {
+  TelemetryPluginStart,
+  TelemetryPluginSetup,
+} from '../../../../../../src/plugins/telemetry/server';
+
+export type SearchTypes =
+  | string
+  | string[]
+  | number
+  | number[]
+  | boolean
+  | boolean[]
+  | object
+  | object[]
+  | undefined;
+
+export interface TelemetryEvent {
+  [key: string]: SearchTypes;
+  '@timestamp'?: string;
+  datastream?: {
+    [key: string]: SearchTypes;
+    dataset?: string;
+  };
+  cluster_name?: string;
+  cluster_uuid?: string;
+  file?: {
+    [key: string]: SearchTypes;
+    Ext?: {
+      [key: string]: SearchTypes;
+    };
+  };
+}
+
+// Copied from telemetry_collection/get_cluster_info.ts
+export interface ESClusterInfo {
+  cluster_uuid: string;
+  cluster_name: string;
+  version: {
+    number: string;
+    build_flavor: string;
+    build_type: string;
+    build_hash: string;
+    build_date: string;
+    build_snapshot?: boolean;
+    lucene_version: string;
+    minimum_wire_compatibility_version: string;
+    minimum_index_compatibility_version: string;
+  };
+}
+
+/**
+ * Get the cluster info from the connected cluster.
+ *
+ * This is the equivalent to GET /
+ *
+ * @param {function} callCluster The callWithInternalUser handler (exposed for testing)
+ */
+export function getClusterInfo(callCluster: LegacyAPICaller) {
+  return callCluster<ESClusterInfo>('info');
+}
+
+interface AllowlistFields {
+  [key: string]: boolean | AllowlistFields;
+}
+
+// Allow list for the data we include in the events. True means that it is deep-cloned
+// blindly. Object contents means that we only copy the fields that appear explicitly in
+// the sub-object.
+const allowlistEventFields: AllowlistFields = {
```

Comment:
This looks good to me. FYI @bfilar @pjhampton @jeska

Reply:
LGTM